### ISSUE TITLE
Fix S2589 FP: Local functions that update captured variables

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#8893]
### Description

S2589: Change this condition so that it does not always evaluate to 'False'. (https://rules.sonarsource.com/csharp/RSPEC-2589)

### Repro steps

```c#
const int desiredOpponents = 1;
var opponents = new List<string>();
var countries = new List<string>
{
    "BY",
    "NL",
};

foreach (var country in countries)
{
    AddOpponents(country);

    if (opponents.Count >= desiredOpponents) // <- warning is reported on this line
    {
        break;
    }
}

void AddOpponents(string country)
{
    opponents.AddRange(new List<string>
    {
        $"{country} 1",
        $"{country} 2",
    });
}
```

### Expected behavior

No warnings are reported.

### Actual behavior

Despite `opponents` collection is updated, analyzer thinks that collection's length will not be changed after calling local function.

### Related information

* C#/VB.NET Plugins version: 9.21.0.86780

**Comment by Tim-Pohlmann** [comment]
Hi @hankovich, thanks for the report!
Unfortunately, our Symbolic Engine does not yet support local functions (#7476). This means that currently, S2589 and other SE rules (e.g., S2583, S2259, and S4158) will not be able to track the application state correctly for code using them.
